src/main/java/org/jruby/ext/openssl/PKeyEC.java

```diff
@@ -931,44 +935,54 @@ private static RaiseException newError(final Ruby runtime, final String message)
-        public IRubyObject initialize(final ThreadContext context, final IRubyObject[] args) {
-            final Ruby runtime = context.runtime;
+        @JRubyMethod(visibility = Visibility.PRIVATE)
+        public IRubyObject initialize(final ThreadContext context, final IRubyObject groupOrPoint) {
+            getPointAndGroup(context, groupOrPoint);
 
-            final int argc = Arity.checkArgumentCount(runtime, args, 1, 2);
-            final IRubyObject arg = args[0];
+            return this;
+        }
 
-            if ( arg instanceof Point ) {
-                this.group = ((Point) arg).group;
-                this.point = ((Point) arg).point;
+        @JRubyMethod(visibility = Visibility.PRIVATE)
+        public IRubyObject initialize(final ThreadContext context, final IRubyObject groupOrPoint, final IRubyObject bn) {
+            if (getPointAndGroup(context, groupOrPoint)) {
                 return this;
             }
 
-            if ( arg instanceof Group ) {
-                this.group = (Group) arg;
+            final byte[] encoded;
+            if (bn instanceof BN) {
+                encoded = ((BN) bn).getValue().abs().toByteArray();
             } else {
-                throw runtime.newTypeError(arg, _EC(runtime).getClass("Group"));
+                encoded = bn.convertToString().getBytes();
             }
-
-            if ( argc == 2 ) { // (group, bn)
-                final byte[] encoded;
-                if (args[1] instanceof BN) {
-                    encoded = ((BN) args[1]).getValue().abs().toByteArray();
-                } else {
-                    encoded = args[1].convertToString().getBytes();
-                }
-                try {
-                    this.point = ECPointUtil.decodePoint(group.getCurve(), encoded);
-                }
-                catch (IllegalArgumentException ex) {
-                    // MRI: OpenSSL::PKey::EC::Point::Error: invalid encoding
-                    throw newError(context.runtime, ex.getMessage());
-                }
+            try {
+                this.point = ECPointUtil.decodePoint(group.getCurve(), encoded);
+            }
+            catch (IllegalArgumentException ex) {
+                // MRI: OpenSSL::PKey::EC::Point::Error: invalid encoding
+                throw newError(context.runtime, ex.getMessage());
             }
 
             return this;
         }
 
+        private boolean getPointAndGroup(ThreadContext context, IRubyObject groupOrPoint) {
+            final Ruby runtime = context.runtime;
+
+            if ( groupOrPoint instanceof Point) {
+                this.group = ((Point) groupOrPoint).group;
+                this.point = ((Point) groupOrPoint).point;
+                return true;
+            }
+
+            if ( groupOrPoint instanceof Group) {
+                this.group = (Group) groupOrPoint;
+                this.point = (ECPoint) ((Group) groupOrPoint).generator(context);
```

Comment:
as hinted by the failing tests, this does not return an `ECPoint` but rather the native (Ruby) `org.jruby.ext.openssl.PKeyEC$Point` instance

Reply:
Fixed by getting paramSpec from the group and using that to get the generator point. Seems to be green now.